include/boost/beast/core/impl/flat_buffer.hpp

```diff
@@ -265,14 +265,29 @@ basic_flat_buffer<Allocator>::
 shrink_to_fit()
 {
     auto const len = size();
+
     if(len == capacity())
         return;
+
     char* p;
     if(len > 0)
     {
         BOOST_ASSERT(begin_);
         BOOST_ASSERT(in_);
-        p = alloc(len);
+#ifndef BOOST_NO_EXCEPTIONS
+        try
+        {
+#endif
+            p = alloc(len);
+#ifndef BOOST_NO_EXCEPTIONS
+        }
+        catch(std::exception&)
```

Comment:
shouldn't this be `const&`?

Reply:
I believe best practice is to catch exceptions by mutable reference. It allows the the exception to be modified and re-thrown at the catch site without slicing (since exceptions are normally caught by their base class).

Reply:
there's no slicing, exceptions are not passed by value, and `const&` informs the reader that the exception is not modified